cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RunnableUtils.java

```diff
@@ -203,6 +209,28 @@ public static void sanityCheckBrokersHavingOfflineReplicasOnBadDisks(List<String
     }
   }
 
+  /**
+   * Gracefully stop the ongoing execution (if any) and wait until the execution stops.
+   *
+   * @param kafkaCruiseControl The Kafka Cruise Control instance.
+   */
+  public static void maybeStopOngoingExecutionToModifyAndWait(KafkaCruiseControl kafkaCruiseControl) {
+    if (kafkaCruiseControl.hasOngoingExecution() && kafkaCruiseControl.modifyOngoingExecution(true)) {
```

Comment:
Looks like there is race condition here. 
If there are two requests try to modify the same ongoing execution.
The first one reaches this method first and wait on ongoing tasks to stop, the second come and `if` will false and it proceeds, then in the end one of the two request may fail with error saying something like "there is ongoing execution"?

Reply:
Indeed in this case one of the executions will fail with `there is ongoing execution` exception; hence, it will guarantee safety. However, it will be wasteful due to creating a new cluster model, and that can be avoided.

Updated the logic and added JavaDoc to explain acceptable / potential scenarios.